test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusProdModeTest.java

```diff
@@ -138,16 +137,8 @@ public class QuarkusProdModeTest
     private boolean clearRestAssuredURL;
 
     public QuarkusProdModeTest() {
-        InputStream appPropsIs = Thread.currentThread().getContextClassLoader().getResourceAsStream("application.properties");
-        if (appPropsIs != null) {
-            customApplicationProperties = new Properties();
-            try (InputStream is = appPropsIs) {
-                customApplicationProperties.load(is);
-            } catch (IOException e) {
-                throw new UncheckedIOException("Failed to load application configuration from "
-                        + Thread.currentThread().getContextClassLoader().getResource("application.properties"), e);
-            }
-        }
+        this.configResourceName = Thread.currentThread().getContextClassLoader()
+                .getResourceAsStream(APPLICATION_PROPERTIES) != null ? APPLICATION_PROPERTIES : null;
```

Comment:
Shouldn't this InputStream be closed at some point?

Reply:
Use `getResource` instead

Reply:
Done.